### ISSUE TITLE
max/min/avg on indexes

### ISSUE THREAD

**Issue by snordrapidimages** [rethinkdb/rethinkdb#4399]
When i try use max/min/avg in rethinkdb data explorer everything is working just fine

```javascript
r.db('test').table('items').max({index: 'id'})
Ok!
```
However, when I run this with rethinkdbdash i got the following error:
```javascript
RqlRuntimeError: Expected type STRING but found OBJECT in
r.db("test").table("items").max({index: "id"})
```
This lead me to try following in data explorer:
```javascript
r.db('test').table('items').max(r.expr({index: 'id'}))

FAILED
RqlRuntimeError: Expected type STRING but found OBJECT in
r.db("test").table("items").max({index: "id"})
```

From what i can see, the bug is not in rethinkdbdash. I base this mostly on the error I get while running r.expr in the data explorer but I could be wrong. 

The performance gains in using max with index is huge so i really like this feature, current workaround is using the official rethinkdb driver.

Thank you!

**Comment by danielmewes** [comment]
@snordrapidimages Thank you for the report. This actually does look like rethinkdbdash isn't handling the optional index argument to `min`/`max`/`avg`.
Pinging @neumino .

Wrapping the `{index: 'id'}` into an `r.expr` changes the meaning of it from an optional argument to a regular argument. So that's why your second query for the Data Explorer doesn't work.

**Comment by snordrapidimages** [comment]
That was my first thought aswell so I tried to locate the error in the driver but didn't find any obvious. So I just wanted to run it through you guys before I file an issue at rethinkdbdash.

**Comment by danielmewes** [comment]
Let's see what @neumino says, but it really looks like this is due to missing support for optional arguments to `max` etc. in rethinkdbdash.

**Comment by neumino** [comment]
Indeed, this was a bug in rethinkdbdash. It's fixed in 2.0.12

**Comment by snordrapidimages** [comment]
Great, thank you
